Amélioration graphique (reprise des actions/boutons)

ggrossetie opened the pull request:
Se base sur #303

ggrossetie committed:
Remove Gatsby in favor of React + Snowpack

ggrossetie committed:
Création du config.json avant npm start et npm run build

ggrossetie committed:
(wip) Refonte graphique

- Ajout d'un composant Button
- Intégration du composant Button sur les actions de la page articles
- Amélioration de l'UI sur l'ajout des tags

ggrossetie committed:
Rend le filtre/recherche insensible à la casse

ggrossetie committed:
Reprise de la création d'un article

- Composant Tag
- Améliore le formulaire de création d'un article

Commit:
Use catch-all for netlify

Commit:
Enable Snowpack React fast refresh

Commit:
Underline links by default

Commit:
Cleanup "Add article" form, and enable autofocus

Commit:
fixup! Enable Snowpack React fast refresh

ggrossetie committed:
Reprise du formulaire de création d'un tag

Commit:
Add icon type of button

Commit:
Revamp articles list buttons

ggrossetie committed:
Reprise de l'écran des books

ggrossetie committed:
Réutilisation de composants entre la page books et articles

ggrossetie committed:
Propage l'action onDoubleClick

ggrossetie committed:
Filtre les articles par tags sur la page des articles

ggrossetie committed:
Propage le changement de titre d'un article (et restaure le onSubmit)

ggrossetie committed:
Utilise un wrap pour que les tags passent à la ligne

antoinentl reviewed (commented):
:+1:

thom4parisot commented on front/gatsby/src/components/Articles.module.scss:
```diff
@@ -140,51 +178,26 @@
   margin-right: 0.5rem;
 }
 
+.filterByTags {
+  display: flex;
+  flex-wrap: wrap;
+  > li:not(:last-child) {
+    margin-right: 1rem;
+    margin-bottom: 0.5rem;
+  }
+  margin-top: 0.25rem;
+  margin-bottom: 0.5rem;
+}
```
```suggestion
.filterByTags {
  display: flex;
  flex-wrap: wrap;
  gap: 0.5rem;
  margin-top: 0.5rem;
  margin-bottom: 1rem;
}
```

ggrossetie replied:
D'après MDN `gap` n'est pas supporté par Safari :sob: 

https://developer.mozilla.org/fr/docs/Web/CSS/gap#compatibilit%C3%A9_des_navigateurs

Après j'avoue que les marges avec `:not(:last-child)` c'est un peu moche... peut être qu'il faudrait basculer sur un grid layout ?

thom4parisot replied:
Ah zut, j'avais zappé :-(

Alors ma proposition tient pas.

thom4parisot commented on front/gatsby/src/components/Articles.module.scss:
```diff
@@ -180,11 +180,13 @@
 
 .filterByTags {
   display: flex;
+  flex-wrap: wrap;
```
`flex-🌯: 🌯;`

ggrossetie replied:
bravo, maintenant j'ai faim :laughing:

Commit:
Use `snowpack@3` code splitting, and dynamically load heavy components

Commit:
Load more performant font files first

Commit:
Load used icons only, to reduce the weight of loaded icons

Commit:
Update dependencies

Possibly, they will be tree-shaked in a nicer fashion

ggrossetie committed:
Reprise de la page credentials

Commit:
Complete "Credentials UI"

Commit:
Style "Share" modal

Commit:
Redesign WriteLeft buttons

Commit:
Redesign WriteLeft buttons

ggrossetie committed:
Ajout des favicon en png 16 et 32px

ggrossetie committed:
Fusionne des blocs Editions et Version dans l'onglet de gauche

- Ajout d'un composant Select
- Fusionne les blocs Editions et Versions
- Améliorer l'ergonomie

thom4parisot commented:
Bug : quand on clique sur "Unlink" (Zotero) dans `/credentials`, ça affiche "Loading".

thom4parisot commented:
![image](https://user-images.githubusercontent.com/138627/107629617-affb2a00-6c62-11eb-9aa3-62a94a56a6eb.png)

![image](https://user-images.githubusercontent.com/138627/107629666-bdb0af80-6c62-11eb-85ce-eb09f3bfe58c.png)


---


Les versions/chapitres sont cliquables (en lecture seule d'abord, un accès au mode "modification" se fait sur interaction) :

![image](https://user-images.githubusercontent.com/138627/107506733-d0fa4700-6b9e-11eb-9ab1-67790a21c6cc.png)

---

Les éléments et comportement sont rendus explicites : 

![image](https://user-images.githubusercontent.com/138627/107506755-dc4d7280-6b9e-11eb-8f18-2c991e8b01e5.png)

![image](https://user-images.githubusercontent.com/138627/107506781-e40d1700-6b9e-11eb-86a8-b30fdf86acf4.png)

![image](https://user-images.githubusercontent.com/138627/107506828-f6875080-6b9e-11eb-82ca-b99b7d2c29dd.png)

---

![image](https://user-images.githubusercontent.com/138627/107506868-04d56c80-6b9f-11eb-91eb-c9415625b2a6.png)

![image](https://user-images.githubusercontent.com/138627/107506891-0ef76b00-6b9f-11eb-8112-8d28de5d1f12.png)

---

L'action de "copier" la référence bibliographique est rendue explicite :

![image](https://user-images.githubusercontent.com/138627/107506975-2f272a00-6b9f-11eb-8f95-1b149b201fdd.png)

"Edit" et "Versions" sont regroupées : 

![image](https://user-images.githubusercontent.com/138627/107507151-701f3e80-6b9f-11eb-967f-8eeb7163437b.png)

La sauvegarde est guidée : 

![image](https://user-images.githubusercontent.com/138627/107507200-83320e80-6b9f-11eb-889f-8862ae9c0da1.png)

On sort plus facilement de la comparaison : 

![image](https://user-images.githubusercontent.com/138627/107507312-ab217200-6b9f-11eb-99c9-f4a7a990119c.png)

---

Les styles sont harmonisés avec les métadonnées

![image](https://user-images.githubusercontent.com/138627/107629960-2435cd80-6c63-11eb-8373-c7ecaef08ef2.png)


La suppression de blocs/mot-clés est plus évidente

![image](https://user-images.githubusercontent.com/138627/107630059-4f202180-6c63-11eb-90cd-8293ae51d6a2.png)

Commit:
Update ui library demo at `/ux`

ggrossetie committed:
Supprime le composant Edit (fusionné avec Versions)

Commit:
Redesign login/register

thom4parisot commented:
@Mogztter 👀 

![image](https://user-images.githubusercontent.com/138627/107538426-43315280-6bc4-11eb-9ea8-5116c4fdf0a4.png)

ggrossetie commented:
Tomme de Savoie overflow:

![tomme-de-savoie](https://user-images.githubusercontent.com/333276/107556227-6f56ce80-6bd8-11eb-901e-dffc8b33fbd2.jpg)

ggrossetie commented:
Je viens aussi de voir qu'il y a un soucis sur les `<textarea>` qui déborde sur le panneau latéral metadata. En supprimant `min-width: 100%;` ça corrige le soucis.

thom4parisot commented:
J'ai ajouté d'autres captures d'écran, et listé les bugs connus en entête de pull request.

Commit:
Fix the Login/ConnectedLogin mixup during merges

Commit:
Rename `YAML.safeLoadAll()` into `YAML.loadAll()`
